controllers/nim_account_controller.go

```diff
@@ -0,0 +1,424 @@
+
+	if cmap, err := r.reconcileNimConfig(ctx, account, apiKeyStr, availableRuntimes); err != nil {
+		logger.V(1).Error(err, "nim configmap reconcile failed")
+		return ctrl.Result{}, err
+	} else {
+		ref, refErr := reference.GetReference(r.Scheme(), cmap)
+		if refErr != nil {
+			return ctrl.Result{}, refErr
+		}
+		targetStatus.NIMConfig = ref
+	}
+	logger.V(1).Info("data config map reconciled successfully")
+
+	// set successful configmap status condition
+	meta.SetStatusCondition(&targetStatus.Conditions, makeConfigMapSuccessfulCondition(account.Generation, ""))
+	// progress account status condition and set initial template status condition
+	meta.SetStatusCondition(&targetStatus.Conditions, makeAccountFailureCondition(account.Generation, "template not reconciled"))
+	meta.SetStatusCondition(&targetStatus.Conditions, makeTemplateFailureCondition(account.Generation, "template not reconciled"))
+
+	if template, err := r.reconcileRuntimeTemplate(ctx, account); err != nil {
+		logger.V(1).Error(err, "runtime template reconcile failed")
+		return ctrl.Result{}, err
+	} else {
+		ref, refErr := reference.GetReference(r.Scheme(), template)
+		if refErr != nil {
+			return ctrl.Result{}, refErr
+		}
+		targetStatus.RuntimeTemplate = ref
+	}
+	logger.V(1).Info("runtime template reconciled successfully")
+
+	// set successful template status condition
+	meta.SetStatusCondition(&targetStatus.Conditions, makeTemplateSuccessfulCondition(account.Generation, ""))
+	// progress account status condition and set initial pull secret status condition
+	meta.SetStatusCondition(&targetStatus.Conditions, makeAccountFailureCondition(account.Generation, "pull secret not reconciled"))
+	meta.SetStatusCondition(&targetStatus.Conditions, makePullSecretFailureCondition(account.Generation, "pull secret not reconciled"))
+
+	if pullSecret, err := r.reconcileNimPullSecret(ctx, account, apiKeyStr); err != nil {
+		logger.V(1).Error(err, "pull secret reconcile failed")
+		return ctrl.Result{}, err
+	} else {
+		ref, refErr := reference.GetReference(r.Scheme(), pullSecret)
+		if refErr != nil {
+			return ctrl.Result{}, refErr
+		}
+		targetStatus.NIMPullSecret = ref
+	}
+	logger.V(1).Info("pull secret reconciled successfully")
+
+	// set successful account and pull secret status conditions
+	meta.SetStatusCondition(&targetStatus.Conditions, makeAccountSuccessfulCondition(account.Generation, ""))
+	meta.SetStatusCondition(&targetStatus.Conditions, makePullSecretSuccessfulCondition(account.Generation, ""))
+
+	return ctrl.Result{}, nil
+}
+
+func (r *NimAccountReconciler) reconcileNimConfig(ctx context.Context, account *v1.Account, apiKey string, runtimes []utils.NimRuntime) (*corev1.ConfigMap, error) {
+	cmap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      constants.NimDataConfigMapName,
+			Namespace: account.Namespace,
+		},
+	}
+
+	if _, err := controllerutil.CreateOrUpdate(ctx, r.Client, cmap, func() error {
```

Comment:
not sure if it is applicable, but eventually you can use Server Side Apply instead of `CreateOrUpdate` which may reduce calls and conflicts

Reply:
Resolved in https://github.com/opendatahub-io/odh-model-controller/pull/289/commits/de002501c908763dce09e79acc6551f2ea8e5551.